src/calendar.rs

```diff
@@ -0,0 +1,190 @@
+            .calendar_hub
+            .calendars()
+            .delete(calendar_id)
+            .doit()
+            .await
+        {
+            error!(?why, "Failed to delete the calendar");
+            return Err(Error::GoogleError(why));
+        };
+        Ok(())
+    }
+
+    #[instrument(skip(self))]
+    pub async fn create_event(&self, event: Event, calendar_id: &str) -> Result<Event, Error> {
+        match self
+            .calendar_hub
+            .events()
+            .insert(event, calendar_id)
+            .doit()
+            .await
+        {
+            Ok(event) => Ok(event.1),
+            Err(why) => {
+                error!(?why, "Failed to create an event");
+                Err(Error::GoogleError(why))
+            }
+        }
+    }
+
+    #[instrument(skip(self))]
+    pub async fn delete_event(&self, id: &str, calendar_id: &str) -> Result<(), Error> {
+        if let Err(why) = self
+            .calendar_hub
+            .events()
+            .delete(calendar_id, id)
+            .doit()
+            .await
+        {
+            error!(?why, "Failed to delete an event");
+            return Err(Error::GoogleError(why));
+        };
+        Ok(())
+    }
+
+    #[instrument(skip(self))]
+    pub async fn list_calendars(&self) -> Result<Vec<CalendarListEntry>, Error> {
+        match self.calendar_hub.calendar_list().list().doit().await {
+            Ok(response) => Ok(response.1.items.unwrap_or_else(|| unreachable!())),
+            Err(why) => {
+                error!(?why, "Failed to list calendars");
+                Err(Error::GoogleError(why))
+            }
+        }
+    }
+
+    #[instrument(skip(self))]
+    pub async fn list_events(&self, calendar_id: &str) -> Result<Vec<Event>, Error> {
+        match self.calendar_hub.events().list(calendar_id).doit().await {
+            Ok(response) => Ok(response.1.items.unwrap_or_else(|| unreachable!())),
+            Err(why) => {
+                error!(?why, "Failed to list events");
+                Err(Error::GoogleError(why))
+            }
+        }
+    }
+
+    #[instrument(skip(self))]
+    pub async fn get_calendars_by_guild_id(
+        &self,
+        guild_id: &GuildId,
+    ) -> Result<Vec<CalendarListEntry>, Error> {
+        let mut response = self.list_calendars().await?;
+        response.retain(|calendar| calendar.summary == Some(guild_id.to_string()));
+        if response.len() > 1 {
+            warn!("More than one calendar associated with the server, using the newest one");
+        };
+        Ok(response)
+    }
+
+    #[instrument(skip(self))]
+    pub async fn get_event_id_by_label(
+        &self,
+        label: &str,
+        calendar_id: &str,
+    ) -> Result<Vec<String>, Error> {
+        let response = self.list_events(calendar_id).await?;
+        let value = response
+            .into_iter()
+            .filter_map(|event| {
+                if event.summary == Some(label.to_string()) {
```

Comment:
Rather than convert a label to a string here, you likely could just compare the &str of event.summary.